tests/unit/amo/utils/test_index.js

```diff
@@ -259,4 +260,48 @@ describe(__filename, () => {
       );
     });
   });
+
+  describe('sendBeacon', () => {
+    it('should send a becaon if navigator.sendBeacon exists', () => {
+      const urlString = 'https://www.mozilla.org';
+      const _log = getFakeLogger();
+      const _navigator = { sendBeacon: sinon.spy() };
+
+      sendBeacon({ _log, _navigator, urlString });
+      sinon.assert.calledWith(_log.debug, `Sending beacon to ${urlString}`);
+      sinon.assert.calledWith(_navigator.sendBeacon, urlString);
+    });
+
+    it('can include data in a becaon', () => {
+      const urlString = 'https://www.mozilla.org';
+      const data = 'some-data';
+      const _log = getFakeLogger();
+      const _navigator = { sendBeacon: sinon.spy() };
+
+      sendBeacon({ _log, _navigator, urlString, data });
+      sinon.assert.calledWith(_navigator.sendBeacon, urlString, data);
+    });
+
+    it('should not send a becaon if navigator does not exist', () => {
+      const urlString = 'https://www.mozilla.org';
+      const _log = getFakeLogger();
+
+      sendBeacon({ _log, _navigator: null, urlString });
+      sinon.assert.calledWith(
```

Comment:
Asserting that `sendBeacon` isn't actually called seems a bit better than asserting a log statement. I feel like we should not have to use `_log` in the function (and only use `log`).

Reply:
I would *love* to be able to assert that `sendBeacon` wasn't called, but the difficulty here is that we're testing a case where `_navigator` doesn't exist, which means that `_navigator.sendBeacon` is not a mock, and therefore there is no way for us to assert that. I agree that relying on log statements in a test is less than ideal, but this seemed like the best way to actually have a test for this, when we don't have a mock we can check.

Reply:
![](https://media1.giphy.com/media/yvBAuESRTsETqNFlEl/giphy.gif)